File: lib/pelican_installer/screens/summary.py
"""Installation summary/completion screen."""

from __future__ import annotations

from textual import on
from textual.app import ComposeResult
from textual.containers import Container, Vertical
from textual.screen import Screen
from textual.widgets import Button, Static

from pelican_installer.utils.state import InstallState


class SummaryScreen(Screen[str]):
    """Final screen showing installation summary."""

    CSS = """
    SummaryScreen {
        align: center middle;
    }

    #summary-container {
        width: 100%;
        height: auto;
        margin-bottom: 1;
        padding: 1;
        border: solid #777777;
        background: #2a2a2a;
    }

    .summary-item {
        margin-bottom: 0;
        color: #e9f6ff;
    }

    .summary-label {
        color: #b7dbe8;
    }

    #next-steps {
        margin-top: 1;
        margin-bottom: 1;
        color: #b7dbe8;
    }
    """

    def __init__(self, state: InstallState) -> None:
        super().__init__()
        self.state = state

    def compose(self) -> ComposeResult:
        with Container(id="root"):
            with Container(id="card"):
                yield Static("Installation Complete! ✓", id="title")

                with Vertical(id="summary-container"):
                    yield Static("INSTALLATION SUMMARY", classes="summary-label")
                    yield Static("")

                    # Component
                    component_name = (
                        "Panel" if self.state.component == "panel" else "Wings"
                    )
                    yield Static(
                        f"✓ Component: {component_name}",
                        classes="summary-item",
                    )

                    # Webserver (if panel)
                    if self.state.component == "panel":
                        yield Static(
                            f"✓ Webserver: {self.state.webserver.capitalize()}",
                            classes="summary-item",
                        )
                        yield Static(
                            f"✓ Protocol: {self.state.protocol.upper()}",
                            classes="summary-item",
                        )
                        yield Static(
                            f"✓ Domain: {self.state.domain}",
                            classes="summary-item",
                        )

                        if self.state.use_ssl:
                            yield Static(
                                f"✓ SSL Certificate: {self.state.domain}",
                                classes="summary-item",
                            )

                    yield Static("")
                    yield Static(
                        "✓ Dependencies installed",
                        classes="summary-item",
                    )
                    yield Static(
                        f"✓ {component_name} configured",
                        classes="summary-item",
                    )

                if self.state.component == "panel":
                    access_url = f"{self.state.protocol}://{self.state.domain}/installer"
                    yield Static(
                        f"Access Panel: {access_url}",
                        id="next-steps",
                    )
                else:
                    yield Static(
                        "Configure Wings through Panel dashboard",
                        id="next-steps",
                    )

                yield Static(
                    "Installation complete! Press Exit to close.",
                    id="hint",
                )

                with Container(id="footer"):
                    yield Button("Exit", id="exit")

    @on(Button.Pressed, "#exit")
    def exit_pressed(self) -> None:
        """Exit the application."""
        self.app.exit()

    def action_request_close(self) -> None:
        """Global close action (c key)."""
        self.app.exit()

File: src/screens/summary.rs
use crate::screens::{Screen, ScreenResult};
use crate::utils::state::{InstallState, ComponentType};
use ratatui::prelude::*;
use ratatui::widgets::Paragraph;
use crossterm::event::KeyEvent;

pub struct SummaryScreen;

impl SummaryScreen {
    pub fn new() -> Self {
        Self
    }
}

impl Screen for SummaryScreen {
    fn render(&mut self, area: Rect, frame: &mut Frame, state: &InstallState) -> crate::error::Result<()> {
        let chunks = Layout::default()
            .direction(Direction::Vertical)
            .margin(1)
            .constraints([
                Constraint::Length(3),  // Title
                Constraint::Min(15),   // Summary
                Constraint::Length(2),  // Next steps
                Constraint::Length(2),  // Hint
                Constraint::Length(3),  // Footer
            ])
            .split(area);
        
        // Title
        let title = Paragraph::new("Installation Complete! ✓")
            .style(Style::default().fg(Color::Green).add_modifier(Modifier::BOLD))
            .alignment(Alignment::Center);
        frame.render_widget(title, chunks[0]);
        
        // Summary content
        let component_name = match state.component {
            Some(ComponentType::Panel) => "Panel",
            Some(ComponentType::Wings) => "Wings",
            None => "Component",
        };
        
        let mut summary_lines = vec![
            "INSTALLATION SUMMARY".to_string(),
            "".to_string(),
            format!("✓ Component: {}", component_name),
        ];
        
        if state.component == Some(ComponentType::Panel) {
            summary_lines.push(format!("✓ Webserver: {}", state.webserver.title()));
            summary_lines.push(format!("✓ Protocol: {}", state.protocol.title()));
            summary_lines.push(format!("✓ Domain: {}", state.domain));
            
            if state.use_ssl {
                summary_lines.push(format!("✓ SSL Certificate: {}", state.domain));
            }
        }
        
        summary_lines.extend_from_slice(&[
            "".to_string(),
            "✓ Dependencies installed".to_string(),
            format!("✓ {} configured", component_name),
        ]);
        
        let summary_text = summary_lines.join("\n");
        let summary_para = Paragraph::new(summary_text)
            .style(Style::default().fg(Color::White))
            .alignment(Alignment::Left);
        frame.render_widget(summary_para, chunks[1]);
        
        // Next steps
        let next_steps = if state.component == Some(ComponentType::Panel) {
            format!("Access Panel: {}://{}/installer", state.protocol.as_str(), state.domain)
        } else {
            "Configure Wings through Panel dashboard".to_string()
        };
        let steps_para = Paragraph::new(next_steps)
            .style(Style::default().fg(Color::Cyan))
            .alignment(Alignment::Center);
        frame.render_widget(steps_para, chunks[2]);
        
        // Hint
        let hint = Paragraph::new("Installation complete! Press Exit to close.")
            .style(Style::default().fg(Color::Gray))
            .alignment(Alignment::Center);
        frame.render_widget(hint, chunks[3]);
        
        // Footer
        let footer = Paragraph::new("Exit (c)")
            .style(Style::default().fg(Color::Yellow))
            .alignment(Alignment::Center);
        frame.render_widget(footer, chunks[4]);
        
        Ok(())
    }
    
    fn handle_key(&mut self, key: KeyEvent, _state: &mut InstallState) -> crate::error::Result<ScreenResult> {
        use crossterm::event::KeyCode;
        
        match key.code {
            KeyCode::Char('c') | KeyCode::Enter => {
                return Ok(ScreenResult::Exit);
            }
            _ => {}
        }
        Ok(ScreenResult::None)
    }
}
